scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/PipelineSubscriber.kt

```diff
@@ -285,12 +288,22 @@ class PipelineSubscriber(
                     pipeline.stop()
                 }
             }
+
+        var pipelineError: PipelineStatus? = null
+        val errTopics = kafkaAdmin.deleteTopics(steps)
+        if (errTopics != null) {
+            pipelineError =
+                PipelineStatus.Error(null)
+                    .withException(errTopics)
+                    .withMessage("kafka streams topic deletion error")
+        }
+
         client.pipelineUpdateEvent(
             makePipelineUpdateEvent(
                 metadata = metadata,
                 operation = PipelineOperation.Delete,
-                success = true,
-                reason = "pipeline removed",
+                success = pipelineError == null,
```

Comment:
A note that `success == false` will transition the Pipeline to "PipelineFailed" on the scheduler side and prevent this pipeline from being removed from k8s. However, in this particular case we have removed the pipeline, and it's just that we weren't able to delete the topics. I think this is fine, as long as we retry deleting the topics when we delete a pipeline already in the "PipelineFailed" state.